src/run/handlers/tracer.cts

```diff
@@ -89,3 +89,16 @@ export function getTracer(): RuntimeTracer {
 
   return tracer
 }
+
+export function recordWarning(warning: Error, span?: Span) {
```

Comment:
`getCacheKeySpan` looked to be non-nullable, can we make `span` non-nullable here as well?

Reply:
https://github.com/opennextjs/opennextjs-netlify/blob/41bca906f838cc7fb1e45ea722db75826422b92e/src/shared/cache-types.cts#L210-L214 this call doesn't have access to `span` to pass it, so that's why I made it optional.

Generally preferably we do pass span explicitly instead of getting "active span", because it's more stable on which span things would be recorded, but it's better to record at all than skip recording because we don't have reference to a span.

Alternative could be to make above linked code get active span and then make `span` here non-nullable, but I did want to simplify usage of it, but would be fine making that change (less "magic")

Reply:
I did attempt the alternative but to me it moved to much tracing related logic and checks to module that doesn't already have reference to containing span I mentioned above and concern of finding active span seems much more in `tracer.cts` module than modules that might want to record warnings, so I'd prefer to keep the signature of this function as it is